src/main/java/org/owasp/validator/html/scan/AntiSamyDOMScanner.java

```diff
@@ -68,6 +68,8 @@
             Pattern.compile("[\\u0000-\\u001F\\uD800-\\uDFFF\\uFFFE-\\uFFFF&&[^\\u0009\\u000A\\u000D]]");
     private static final Pattern conditionalDirectives =
             Pattern.compile("<?!?\\[\\s*(?:end)?if[^]]*\\]>?");
+    private static final Pattern styleUnwantedImport =
+            Pattern.compile("@import(\\s)*((\'|\")?(\\s)*(http://|https://)?([^\\s;]*)(\\s)*(\'|\")?(\\s)*;?)", Pattern.CASE_INSENSITIVE);
```

Comment:
regex can be optimized
**(http://|https://)** ==> **(https?://)**
also don't we need to escape **//** like **\/\/**, it's required in javascript not sure about java though

Reply:
in some cases this is not matching the entire string, check below screenshot
![Screenshot 2021-12-22 at 3 43 50 PM](https://user-images.githubusercontent.com/619792/147076534-845152b4-d2a1-4cb7-9f87-7991db48f618.png)

Reply:
I recreated regex in the following way:
1. consider pair of double-quotes `""`
`@import(\s)*[^;\"]*(\".*?\")[^;]*?;+?`

2. consider a case that two or more pairs of quoble-quotes
`@import(\s)*([^;\"]*(\".*?\")[^;]*?)+?;+?`

3. consider a case with no `;`
`@import(\s)*([^;\"]*(\".*?\")[^;]*?)+?(;+?|$)`

4. consider pair of single quotes `''` in the same way
`@import(\s)*([^']*('.*?')[^;]*?)+?(;+?|$)`

5. integrate 3 and 4
`@import(\s)*([^;\"']*((\".*?\")|('.*?'))[^;]*?)+?(;+?|$)`

6. consider pair of parentheses `()` in the same way
`@import(\s)*([^;\"'\(]*((\".*?\")|('.*?')|(\(.*?\)))[^;]*?)+?(;+?|$)`

7. consider a case with no quote, double-quote or parenthesis
`@import(\s)*[^;\"'\(]*?(;+?|$)`

8. integrate 6 and 7
`@import(\s)*(([^;\"'\(]*((\".*?\")|('.*?')|(\(.*?\)))[^;]*?)+?|([^;\"'\(]*?))(;+?|$)`

9. espace `\s`, `\(` and `\)`
`@import(\\s)*(([^;\"'\\(]*((\".*?\")|('.*?')|(\\(.*?\\)))[^;]*?)+?|([^;\"'\\(]*?))(;+?|$)`